internal/infra/http/server.go

```diff
@@ -26,45 +26,42 @@ const (
 )
 
 type Component struct {
-	config config
+	config Config
```

Comment:
everything else is a pointer in this struct, as you will be passing it to many other components, maybe use a pointer there too as well? wdyt?

Reply:
Good idea, I updated it.